src/js/tabs/osd.js

```diff
@@ -149,6 +163,128 @@ FONT.openFontFile = function($preview) {
+          for (var x = 0, X = canvas.width; x < X; x++) {
+            var rgbPixel = ctx.getImageData(x, y, 1, 1).data.slice(0, 3),
+                colorKey = rgbPixel.join("-");
+            if (!FONT.constants.LOGO.MCM_COLORMAP[colorKey]) {
+              reject(i18n.getMessage("osdSetupReplaceLogoImageColorsError"));
+              return;
+            }
+          }            
+        }
+        resolve();
+      });
+    };
+
+    chrome.fileSystem.chooseEntry({type: 'openFile', accepts: [{extensions: ['png', 'bmp']}]}, function(fileEntry) {
+      if (chrome.runtime.lastError) {
+          console.error(chrome.runtime.lastError.message);
+          return;
+      }
+      var img = new Image();
+      img.onload = function() {
+        validateImage(img).then(function() {
+          resolve(img);
+        }).catch(function(error) {
+          console.error(error);
+          reject(error);
+        });
+      };
+      img.onerror = function(error) {
+        reject(error);
+      };
+      fileEntry.file(function(file) {
+        img.src = "file://" + file.path;
+      });
+    });
+  });
+};
+
+// replaces the logo in the font based on an Image object
+FONT.replaceLogoFromImage = function(img) {
+  // takes image data from an ImageData object and returns an MCM symbol as an array of strings
+  var imageToCharacter = function(data) {
+    var char = [],
+        line = "";
+    for (var i = 0, I = data.length; i < I; i += 4) {
+      var rgbPixel = data.slice(i, i + 3),
+          colorKey = rgbPixel.join("-");
+      line += FONT.constants.LOGO.MCM_COLORMAP[colorKey]
+        || FONT.constants.LOGO.MCM_COLORMAP['default'];
+      if (line.length == 8) {
+        char.push(line);
+        line = "";
+      }
+    }
+    var fieldSize = FONT.constants.SIZES.MAX_NVM_FONT_CHAR_FIELD_SIZE;
+    if (char.length < fieldSize) {
+      var pad = FONT.constants.LOGO.MCM_COLORMAP['default'].repeat(4);
+      for (var i = 0, I = fieldSize - char.length; i < I; i++)
+        char.push(pad);
+    }
+    return char;
+  };
+
+  // takes an OSD symbol as an array of strings and replaces the in-memory character at charAddress with it
+  var replaceChar = function(lines, charAddress) {
+    var characterBits = [];
+    var characterBytes = [];
+    for (var n = 0, N = lines.length; n < N; n++) {
+      var line = lines[n];
+      for (var y = 0; y < 8; y = y + 2) {
+        var v = parseInt(line.slice(y, y+2), 2);
+        characterBits.push(v);
+      }
+      characterBytes.push(parseInt(line, 2));
+    }
+    FONT.data.characters[charAddress] = characterBits;
+    FONT.data.characters_bytes[charAddress] = characterBytes;
+    FONT.data.character_image_urls[charAddress] = null;
+    FONT.draw(charAddress);
+  };
+
+  // loop through an image and replace font symbols
+  var canvas = document.createElement('canvas'),
+      ctx = canvas.getContext('2d'),
+      charAddr = SYM.LOGO;
+  canvas.width = img.width;
```

Comment:
Indentation.
In general, betaflight-configurator uses 4 spaces. Please fix your code.

Reply:
Sure, no problem! Does this also mean that the code indented with 2 spaces is legacy and was inherited from Cleanflight? Unfortunately I didn't take the time to read through the codebase and commit history, I've only read through the files that are part of this changeset. But I guess that will happen eventually.